src/maestro/agents/openai_agent.py

```diff
@@ -242,6 +332,64 @@ def _process_agent_result(self, result: Optional[Any]) -> str:
             self.print(f"DEBUG [OpenAIAgent {self.agent_name}]: {fallback_str}")
             return fallback_str
 
+    def _extract_token_usage_from_result(self, result: Any) -> None:
+        """Try to extract token usage from the result object."""
+        try:
+            if hasattr(result, "usage"):
+                usage = result.usage
+                if hasattr(usage, "prompt_tokens"):
+                    self.prompt_tokens = usage.prompt_tokens
+                if hasattr(usage, "completion_tokens"):
+                    self.response_tokens = usage.completion_tokens
+                if hasattr(usage, "total_tokens"):
+                    self.total_tokens = usage.total_tokens
+                self.print(
+                    f"INFO [OpenAIAgent {self.agent_name}]: Extracted token usage from result - Prompt: {self.prompt_tokens}, Response: {self.response_tokens}, Total: {self.total_tokens}"
+                )
+                return
+
+            if hasattr(result, "messages"):
+                for message in result.messages:
```

Comment:
I'd move these massive if checks into their own private method. And if these are similarly used other places then put in a utility class or file in utils package that could be reused. Basically search code base for duplicates or similar code